main/src/com/google/refine/expr/functions/Cross.java

```diff
@@ -71,22 +84,27 @@ public Object call(Properties bindings, Object[] args) {
 
                 if (v instanceof WrappedCell) {
                     return lookup.getRows(((WrappedCell) v).cell.value);
+                } else if (v instanceof Number && targetColumnName.equals(INDEX_COLUMN_NAME)){
+                    return lookup.getRows(((Number) v).intValue());
                 } else {
                     return lookup.getRows(v);
                 }
             }
         }
-        return new EvalError(ControlFunctionRegistry.getFunctionName(this) + " expects a cell or cell value, a project name to look up, and a column name in that project");
+        return new EvalError(ControlFunctionRegistry.getFunctionName(this) + " expects a cell or cell value, a project name to look up (optional), and a column name in that project (optional)");
     }
 
     @Override
     public String getDescription() {
-        return "Looks up the given value in the target column of the target project, returns an array of matched rows, cell will be interpreted as cell.value";
+        return "Looks up the given value in the target column of the target project, returns an array of matched rows. " +
+                "The first argument will be interpreted as cell.value if set to cell. " +
+                "The second argument will be interpreted as the current project name if omitted or set to \"\". " +
+                "The third argument will be interpreted as the index (starts from 0) column if omitted or set to \"\"";
     }
 
     @Override
     public String getParams() {
-        return "cell c or object value, string projectName, string columnName";
+        return "cell c or object value, string projectName (optional), string columnName (optional)";
     }
```

Comment:
I would change `object value` for just value everywhere in the doc. It's just a value, it can come from anywhere (even literals values are accepted!). Also, the use of object is misleading, not pertinent and confusing.

Reply:
@antoine2711 Good point! I'll change it. The "jsonize()" function also uses "value" , so I think it's ok to do the same here.

![jsonize](https://user-images.githubusercontent.com/29347603/79676718-76f95100-821b-11ea-8288-47fd0a06bc67.PNG)

Reply:
`value` is loosely defined here: https://github.com/OpenRefine/OpenRefine/wiki/Variables#variables

We probably should give a small mention of `value` somewhere here:
https://github.com/OpenRefine/OpenRefine/wiki/Server-Side-Architecture#data-model
>raw data: a list of rows, each row consisting of a list of cells

Reply:
In the previous version of `cross()` even a value was somehow invisibly « bounded » to the column it came from. So, to name it « object value », or « cell value », had a certain logic in it. But now, it's really a pure value, that can be anything. So, might as well just call it that, a value.

@thadguidry: yah, we have a few definition here and there, so a nice clean up in our new documentation will surely come handy. my favorite is datatype definitions in the Cell page… But hey! At one point, it made a lot of sens, I guess. ;-)

I'm eager to see what's coming in that way.

Regards, Antoine.